### PR TITLE
MP-326 Function helpers on Account class's account object

### PR DESCRIPTION
**Pull request by everton0991**
Scope:
• Abstracts the account object creation.
• Adjust expected error messages for tests.
• Unify Account's tests.
• Get public keys and signatures programmatically for account and multiSig suites.

### CODE REVIEW

File: tests/unit/validatorUtil.js [hunk 1]
```diff
@@ -30,15 +30,15 @@ describe('Unit/Util/Validator:', () => {
...
         
         it('Fail: hexToTxt throws an error if Hex is invalid', () => {
-            expect(() => Validator.hexToTxt(Number('notAHex'))).to.throw('Validator.hexToTxt: TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received type number (NaN)');
+            expect(() => Validator.hexToTxt(Number('notAHex'))).to.throw('The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received type number (NaN)');
         });
         
         it('Fail: txtToHex throws an error if string is invalid', () => {
-            expect(() => Validator.txtToHex(Number('notAHex'))).to.throw('Validator.txtToHex: TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received type number (NaN)');
+            expect(() => Validator.txtToHex(Number('notAHex'))).to.throw('The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received type number (NaN)');
         });
```
Comment (everton0991) [inline tests/unit/validatorUtil.js:42]: Adjust the error messages for the tests above.

File: tests/unit/txUtil.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ describe('Unit/Util/Tx:', () => {
         it('Fail: Reject when called with invalid arguments', async () => {
             await expect(
                 Tx.remainingDeposit(null, null)
-            ).to.eventually.be.rejectedWith('Transaction.rewardDeposit: TypeError: Cannot read property \'DSLinker\' of null');
+            ).to.eventually.be.rejectedWith('Cannot read properties of null (reading \'DSLinker\')');
```
Comment (everton0991) [inline tests/unit/txUtil.js:37]: Adjust error message

File: src/Account.js [hunk 3]
```diff
@@ -44,17 +85,14 @@ class Accounts {
                     throw "Account already added"
                 }
             }
-            let utxo = { "DataStores": [], "ValueStores": [], "AtomicSwaps": [], "ValueStoreIDs": [], "DataStoreIDs": [], "AtomicSwapIDs": [], "Value": "" }
-            let acct = { "UTXO": utxo, "curve": curve, "address": address, "signer": signer };
-            this.accounts.push(acct);
-            return acct;
+            const account = this._buildAccountObject(curve, address, signer);
+            return account;
```
Comment (ACatThatPrograms) [inline src/Account.js:89]: Nice :100:

File: src/Account.js [hunk 4]
```diff
@@ -15,6 +15,47 @@ class Accounts {
...
+            "getAccountValueStores": async (minValue) => this._getAccountValueStores(address, minValue), 
+            "getAccountDataStores": async (minValue) => {
+                let dataStoreUTXOs = await this.Wallet.Rpc.getDataStoreUTXOIDsAndIndices(address, curve, minValue, false);
+                return dataStoreUTXOs;
+            },
+            "getAccountBalance": async () => {
+                let [,balance] = await this.Wallet.Rpc.getValueStoreUTXOIDs(address, curve, false);
+                return balance;
+            }
+        };
```
Comment (ACatThatPrograms) [inline src/Account.js:53]: :fire:

File: src/Account.js [hunk 5]
```diff
@@ -147,7 +183,6 @@ class Accounts {
         try {
             address = this.Wallet.Utils.isAddress(address)
             let accountIndex = await this._getAccountIndex(address)
-            this.accounts[accountIndex]["UTXO"] = { "DataStores": [], "ValueStores": [], "AtomicSwaps": [], "ValueStoreIDs": [], "DataStoreIDs": [], "AtomicSwapIDs": [], "Value": "" }
```
Comment (ACatThatPrograms) [inline src/Account.js:150]: Why are we removing this? , iirc it clears the UTXO state before updating it below so it's accurate

File: src/Account.js [hunk 6]
```diff
@@ -178,7 +213,6 @@ class Accounts {
             }
             address = this.Wallet.Utils.isAddress(address)
             let accountIndex = await this._getAccountIndex(address)
-            this.accounts[accountIndex]["UTXO"] = { "DataStores": [], "ValueStores": [], "AtomicSwaps": [], "ValueStoreIDs": [], "DataStoreIDs": [], "AtomicSwapIDs": [], "Value": "" }
```
Comment (ACatThatPrograms) [inline src/Account.js:181]: Same:
`Why are we removing this? , iirc it clears the UTXO state before updating it below so it's accurate`

File: src/Account.js [hunk 7]
```diff
@@ -200,11 +234,15 @@ class Accounts {
...
+    /**
+     * Get Value Stores for account
+     * @param {hex} address
+     * @param {number} minValue
+     */
     async _getAccountValueStores(address, minValue) {
         try {
             address = this.Wallet.Utils.isAddress(address)
             let accountIndex = await this._getAccountIndex(address)
-            this.accounts[accountIndex]["UTXO"] = { "DataStores": [], "ValueStores": [], "AtomicSwaps": [], "ValueStoreIDs": [], "DataStoreIDs": [], "AtomicSwapIDs": [], "Value": "" }
```
Comment (ACatThatPrograms) [inline src/Account.js:207]: And again, maybe I'm missing something

File: tests/integration/account.js [hunk 8]
```diff
@@ -12,10 +12,157 @@ describe('Integration/Account:', () => {
...
+            await madWalletInstance.Account.addAccount(privateKey, 1);
+            let msg = "0xc0ffee";
+            let wrongVerifyMsg = "0xcafee";
+            let signature = await madWalletInstance.Account.accounts[0]["signer"].sign(msg);
+            await expect(
+                madWalletInstance.Account.accounts[0]["signer"].verify(wrongVerifyMsg, signature)
+            ).to.eventually.be.rejectedWith('Public Keys don\'t match');
+        })
+    });
+
```
Comment (ACatThatPrograms) [inline tests/integration/account.js:165]: Nicee, thank you for adding test cases. :100:

File: tests/unit/multiSig.js [hunk 9]
```diff
@@ -8,23 +8,32 @@ const MultiSig = require("../../src/Signers/MultiSig");
...
-        '0928b847ee0c4e5851ea6e9943dc8fd3c0cd463e35844fcac6ec303cf7c6796d137bdb1da527b9b822be3d14c866081857a6711e98807b112d3d77bfd434c3be02ed582a08c08c7ddb5355c7f742c1581288e9c43552b16b7ffa16b826d2839116c6c3bea6c0cef1b38088c4ab31daa51eade02a9b8100e04e7de7d88b7b8fb9'
-    ];
-    const signatures = [
-        '14909a6ab1c19ad26264cb1491885a8ba9a936264ab7c71d4cc58f889419b458289da71e04086b804a4b74b358ef923e2f6cba9bf87310724d72571efcd49e0802390d2d1dc0715f9a7d0f36d5b511baf5e0833bf3f0db881457ce5b523e1d0512dba7a1a0d623805fc79c642efb77fd3286747bbaa71c4365fa3f64a80fc83c2cbccc0e2f548ee60fc5f6cb407b63b281be07068cd3b4dbc6f7aaf983471d2318f1bc95df4b1bb3cf82d2e45b64b9d461b902c59b8d85c9b71761445b72b37d', 
-        '14909a6ab1c19ad26264cb1491885a8ba9a936264ab7c71d4cc58f889419b458289da71e04086b804a4b74b358ef923e2f6cba9bf87310724d72571efcd49e0802390d2d1dc0715f9a7d0f36d5b511baf5e0833bf3f0db881457ce5b523e1d0512dba7a1a0d623805fc79c642efb77fd3286747bbaa71c4365fa3f64a80fc83c2cbccc0e2f548ee60fc5f6cb407b63b281be07068cd3b4dbc6f7aaf983471d2318f1bc95df4b1bb3cf82d2e45b64b9d461b902c59b8d85c9b71761445b72b37d'
-    ];
```
Comment (ACatThatPrograms) [inline tests/unit/multiSig.js:20]: Yeah this is good to remove -- Burned public ones already, but magic values make for a bad time.

**Review by ACatThatPrograms — COMMENTED** [review]
Looks good just the few questions before merging

:100: on the test cases

**Pull request merged**